src/Core/Resolvers/QueryExecutor.cs

```diff
@@ -636,6 +645,39 @@ DbResultSet dbResultSet
             return resultProperties;
         }
 
+        /// <summary>
+        /// Reads data into jsonString.
+        /// </summary>
+        /// <param name="dbDataReader">DbDataReader.</param>
+        /// <param name="availableSize">Available buffer.</param>
+        /// <param name="jsonString">jsonString to read into.</param>
+        /// <returns>size of data read in bytes.</returns>
+        public int StreamData(DbDataReader dbDataReader, int availableSize, StringBuilder jsonString)
+        {
+            int fieldSize = (int)dbDataReader.GetChars(0, 0, null, 0, 0);
+
+            // if the size of the field is less than available size, then we can read the entire field.
+            // else we throw exception.
+            ValidateSize(availableSize, fieldSize);
+
+            char[] buffer = new char[fieldSize];
```

Comment:
do we need to wrap this in try/catch to gracefully handle OOM case?

Reply:
if dab is processing >1 request, `availableSize` may be out of date by the time you're creating your buffer and could result in buffer alloc failure, couldn't it?

Reply:
Available size is the size available per request. So we are not keeping track of it across requests. You would have to have separate throttling limits for high cost requests.

Reply:
availableSize is on a per request basis. We are trying to limit the amount of data that can be pulled in a single request. I think solution for multiple requests would be to appropriate throtting of requests at api layer and then appropriately scaling instances.

Reply:
It's still possible for ` char[] buffer = new char[fieldSize];` to raise an exception though.

Reply:
but i suppose we don't want to be wrapping all `new` usage in try/catch

Reply:
It would be very unlikely for buffer to raise an OOM exception though as we validate against the 158MB limit set (in the worst case) in validate size call and the memory should be available for GC after this function finishes.